Adding a li2Xliff guide to the documentation

JSchenk8 opened the pull request:
We have built our installable li2Xliff guide to offer CAT translations for customers. This is implemented by the NZZ. 

I have tried to write as simple a guide as possible but with enough detail. There are links to the webhook and access-hook docs as these are required to implement it. I have given example functions for these, but otherwise kept it simple on the configs you can hand li2xliff and what these configs do.

JSchenk8 commented:
@DaRaFF @looer @meinradandermatt  I struggled to find the balance between enough and too much detail here. Could happily add more, but hopefully it is clear enough for other developers to get started. Let me know what you think, very happy to revisit the depth.

meinradandermatt commented on content/guides/integrations/li2xliff/index.md:
```diff
@@ -0,0 +1,145 @@
+
+## Registering Translation Hooks
+
+In your Livingdocs Server instance you can register a translation [access-hook]({{< ref "/guides/authentication/access-hooks.md" >}}) which can then send your document for translation:
```
why an access hook? We use the server events.

JSchenk8 replied:
I was confused about registering the hook which also listens to the server event. Would you say we could simplify this guide with just a link to the sever events docs? (here)[https://docs.livingdocs.io/reference-docs/server-extensions/server-events/]

meinradandermatt replied:
yes that make sense

looer commented:
I think you can add everything from the [Readme](https://github.com/livingdocsIO/li2xliff) to the guide as well.

Then would be cool to have an example of the full implementation, removing everything from NZZ and simplifying it a little, because customer are going to ask questions about how to have it fully working in the downstream.

JSchenk8 commented:
@meinradandermatt  I've updated the server subscription section. I added a link to the initialized hook docs after discussion with @looer . What do you think about adding a full implementation guide? Or is this enough?

DaRaFF commented on content/reference-docs/project-config/settings.md:
```diff
@@ -277,6 +277,8 @@ Livingdocs allows you to define documents in multiple languages as well as integ
 
 If you activate the translation feature, then the `language` metadata plugin on a document will contain a `groupId`. You can use this in the `languageGroupId` parameter of the [Publication Search public API](https://edit.livingdocs.io/public-api) (under "Search Publications") call to retrieve all translations of a document.
 
+If you want to use a computer assisted translation (CAT) tool to translate livingdocs content you can install our li2xliff library to convert documents to XLIFF for translation and register server hooks to update content. You can follow our in-depth guide [here](add guide link)
```
add a link to
- "server hooks"
- "here" add guide link

JSchenk8 replied:
@DaRaFF Great thank you, I've added these and updated the format 🚀

DaRaFF commented on content/guides/integrations/li2xliff/index.md:
````diff
@@ -0,0 +1,150 @@
+It can be extended with the following optional configs:
+
+```js
+excludeEditables: [{
````
formatting looks not consistent

DaRaFF commented on content/guides/integrations/li2xliff/index.md:
````diff
@@ -0,0 +1,150 @@
+`createXliff` converts Livingdoc content to XLIFF and requires Livingdoc content and a config object:
+
+```js
+  const xliffContent = createXliff({
````
formatting looks not consistent

DaRaFF replied:
I think that's easier to read and to copy.

```js
const xliffContent = createXliff(
  {
    content: documentVersion.content, // Livingdocs Content
    config: {

      // required
      targetLanguage: config.targetLanguage,
      // required
      srcLanguage: config.srcLanguage,

      // optional
      excludeEditables: [
        {
          component: 'title',
          // this will not translate title or author editables inside title components
          editables: ['title', 'author']
        },
        {
          component: 'header',
          // this will not translate titles inside header components
          editables: ['title']
        }
      ],
      // optional - this will translate content inside of strong and anchor tags, but will not preserve the formatting
      excludeTags: ['strong', 'a']
    }
  }
)
```

looer commented:
Some suggestions:
- Add the instruction for installing using `npm install` like in the Readme
- Add link to the library on Github
- Add a mention of what is a CAT tool and some examples with links. Ex Phrase/Memsource, Smartcat, ...
- Update the title to be more homogenous with the other guides. from "Using li2xliff and CAT tools for translating Livingdocs content" to "Assisted translation with CAT tool", and keep the subtitle the same.
![image](https://user-images.githubusercontent.com/1682803/211839944-8e509089-965a-4b6c-9a4b-32db64889cc3.png)

JSchenk8 commented:
@looer Thanks very much! I changed it to reflect your advice, but I didn't recommend a CAT tool as I feel it's not up to us to guide them towards any particular tools, although we can of course say if they ask that "we have set it up with Memsource". Much appreciated 🙏

livingdocs-automation commented:
:tada: This PR is included in version 1.68.153 :tada:

The release is available on [GitHub release](https://github.com/livingdocsIO/documentation/releases/tag/v1.68.153)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
